src/app/screens/ledger/addStxAddress/index.tsx

```diff
@@ -157,6 +162,106 @@ function AddStxAddress(): JSX.Element {
     }
   };
 
+  const renderContent = () => {
+    switch (currentStep) {
+      case Steps.ConnectLedger:
+        return (
+          <LedgerConnectionView
+            title={t('LEDGER_CONNECT.STX_TITLE')}
+            text={t('LEDGER_CONNECT.STX_SUBTITLE')}
+            titleFailed={t('LEDGER_CONNECT.TITLE_FAILED')}
+            textFailed={t('LEDGER_CONNECT.STX_SUBTITLE_FAILED')}
+            imageDefault={ledgerConnectStxIcon}
+            isConnectSuccess={isConnectSuccess}
+            isConnectFailed={isConnectFailed}
+          />
+        );
+      case Steps.VerifyAddress:
+        if (isConnectFailed || isAddressRejected) {
+          return (
+            <LedgerFailView
+              title={t(
+                isAddressRejected
+                  ? 'LEDGER_ADD_ADDRESS.TITLE_CANCELLED'
+                  : 'LEDGER_CONNECT.TITLE_FAILED',
+              )}
+              text={t(
+                isAddressRejected
+                  ? 'LEDGER_ADD_ADDRESS.SUBTITLE_CANCELLED'
+                  : 'LEDGER_CONNECT.BTC_SUBTITLE_FAILED',
+              )}
+            />
+          );
+        }
```

Comment:
should this be its own step?

Reply:
The step code itself is below, and the code above is basically the error handling